client/src/AuthService.js

```diff
@@ -25,36 +33,32 @@ const getToken = () => {
 };
 
 const logout = () => {
-  localStorage.removeItem("token");
+  localStorage.clear();
+  window.location.href = "/signin";
 };
 
-const requestAPI = (url, options) => {
-  const headers = {
-    Accept: "application/json",
-    "Content-Type": "application/json"
-  };
-
-  // Setting Authorization header
-  if (loggedIn()) {
-    headers["Authorization"] = "Bearer " + getToken();
-  }
+const authFetch = (url, options) => {
+  headers["Authorization"] = "Bearer " + getToken();
 
   return fetch(url, {
     headers,
     ...options
-  })
-    .then(_checkStatus)
-    .then(response => response.json());
+  }).then(_checkStatus);
 };
 
 const _checkStatus = response => {
-  if (response.status >= 200 && response.status < 300) {
-    return response;
-  } else {
-    var error = new Error(response.statusText);
-    error.response = response;
-    throw error;
-  }
+  return response.json().then(res => {
```

Comment:
If it's an actual unexpected error, the response won't be in JSON, so this will throw an exception

Reply:
wait nvm

Reply:
looks good actually, you can merge